File: jja6312/2월18일/의상.java
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> map = new HashMap<>();

        // 옷을 종류별로 분류하여 개수를 센다.
        for (String[] cloth : clothes) {
            String type = cloth[1];
            map.put(type, map.getOrDefault(type, 0) + 1);
        }

        // 각 옷 종류별로 입거나 입지 않는 경우의 수를 계산한다. (옷의 개수 + 1)
        int answer = 1;
        for (int count : map.values()) {
            answer *= (count + 1);
        }

        // 아무것도 입지 않는 경우의 수 1을 뺀다.
        return answer - 1;
    }
}

File: jeong-hyuk/2월18일/의상.py
from collections import defaultdict

def solution(clothes):
    closet = defaultdict(int)
    #의상의 종류별로 개수를 셈
    for cloth, kind in clothes:
        closet[kind] += 1
    
    #의상을 선택하는 경우의 수를 구함
    answer = 1
    for count in closet.values():
        #해당 종류의 의상을 입지 않는 경우도 고려하여 +1
        answer *= (count + 1)
    
    #모든 의상을 입지 않는 경우를 제외함
    return answer - 1